File: AdoptaMe_Spring/AdoptaMe/src/main/java/mx/edu/unpa/apidroid/dtos/requests/MascotaRequest.java
package mx.edu.unpa.apidroid.dtos.requests;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Data
public class MascotaRequest {

    @NotNull
    private Long idUsuarioDonador;

    @NotNull
    private Long idTipoMascota;

    @NotBlank
    private String nombre;

    private String raza;

    @NotBlank
    private String sexo;

    private String edadAproximada;

    private String descripcion;

    private String estadoAdopcion;
}

File: AdoptaMe_Android/MiAndroid/app/src/main/java/mx/edu/unpa/miandroid/model/MascotaRequest.kt
package mx.edu.unpa.miandroid.model

data class MascotaRequest(
    val idUsuarioDonador: Long,
    val idTipoMascota: Long,
    val nombre: String,
    val raza: String? = null,
    val sexo: String,
    val edadAproximada: String? = null,
    val descripcion: String? = null,
    val estadoAdopcion: String? = "Disponible"
)